### PR TITLE
Upgrade and satisfy Eastwood

### PR DESCRIPTION
**Pull request by vemv**
https://github.com/jonase/eastwood/blob/master/changes.md#changes-from-0314-to-040

### CODE REVIEW

**Comment by vemv** [comment]
Green in my account

image

File: project.clj [hunk 1]
```diff
@@ -138,9 +138,8 @@
                                           try-if-let [[:block 1]]}}}]
 
              :eastwood [:test
-                        {:plugins [[jonase/eastwood "0.3.11"]]
+                        {:plugins [[jonase/eastwood "0.4.0"]]
                          :eastwood {:config-files ["eastwood.clj"]
-                                    ;; TODO: Add :test-paths once
-                                    ;; https://github.com/jonase/eastwood/issues/298
```
Comment (vemv) [inline project.clj:144]: https://github.com/jonase/eastwood/issues/298 resolved

File: project.clj [hunk 2]
```diff
@@ -138,9 +138,8 @@
...
-                                    ;; is resolved
-                                    :namespaces [:source-paths]}}]})
+                                    :exclude-namespaces [cider.nrepl.middleware.test-filter-tests]
+                                    :ignored-faults {:unused-ret-vals-in-try {cider.nrepl.middleware.profile-test [{:line 25}]}
```
Comment (vemv) [inline project.clj:144]: New option, allows to use e.g. side-effectful forms without causing an Eastwood fault

https://github.com/jonase/eastwood#ignored-faults

File: test/clj/cider/nrepl/middleware/slurp_test.clj [hunk 3]
```diff
@@ -6,10 +6,11 @@
    [clojure.string :as str]))
 
 (t/deftest test-project-clj-is-clj
-  (let [resp (slurp-url-to-content+body
-              (.toString
-               (.toURL
```
Comment (vemv) [inline test/clj/cider/nrepl/middleware/slurp_test.clj:11]: .toURL was deprecated

File: test/clj/cider/nrepl/middleware/stacktrace_test.clj [hunk 4]
```diff
@@ -153,20 +153,21 @@
...
-        :line 1
-        :column 42}
-       (extract-location {:class "clojure.lang.Compiler$CompilerException"
-                          :location {:clojure.error/line 1
-                                     :clojure.error/column 42
-                                     :clojure.error/source "/foo/bar/baz.clj"
-                                     :clojure.error/phase :macroexpand
-                                     :clojure.error/symbol 'clojure.core/let}
-                          :message "Syntax error macroexpanding clojure.core/let at (1:1)."}))))
+    (is (= {:class    "clojure.lang.Compiler$CompilerException"
```
Comment (vemv) [inline test/clj/cider/nrepl/middleware/stacktrace_test.clj:156]: missing `is` + outdated expectation

File: test/clj/cider/nrepl/middleware/util/instrument_test.clj [hunk 5]
```diff
@@ -178,6 +178,5 @@
   ;; This is because Namespace objects implement IMeta, but not IObj.
   (let [an-ns (create-ns (with-meta 'foo.bar {:doc "ns docstring"}))]
     (binding [*ns* an-ns]
-      (t/breakpoint-tester '(let [x (cider.nrepl.middleware.util.instrument-test/ns-embedding-macro)] x))
-      ;; If the above did not throw, then we pass
-      (is true))))
```
Comment (vemv) [inline test/clj/cider/nrepl/middleware/util/instrument_test.clj:183]: Eastwood doesn't like `is true` much

File: test/clj/cider/nrepl/version_test.clj [hunk 6]
```diff
@@ -1,4 +1,4 @@
-(ns cider.nrepl.middleware.version-test
```
Comment (vemv) [inline test/clj/cider/nrepl/version_test.clj:1]: file -> ns mismatch

File: test/spec/cider/nrepl/middleware/stacktrace_spec_test.clj [hunk 7]
```diff
@@ -30,15 +30,15 @@
 (s/def ::person (s/keys :req [::first-name ::last-name ::email]
                         :opt [::phone]))
 
-(deftest spec-assert-stacktrace-test
+(def broken-musk {::first-name "Elon"
```
Comment (vemv) [inline test/spec/cider/nrepl/middleware/stacktrace_spec_test.clj:33]: def-in-def

**Comment by bbatsov** [comment]
Thanks for tackling this! 🙇‍♂️ 

 If you're in the mood for more - you can do the same for some of the other projects as well (e.g. nREPL, Orchard, refactor-nrepl, etc). :-)

**Comment by vemv** [comment]
🍻!

> If you're in the mood for more - you can do the same for some of the other projects as well (e.g. nREPL, Orchard, refactor-nrepl, etc).

Sounds good! Will give it a try soon enough

**Pull request merged**